### PR TITLE
Add 'blank=True' to version license and prevent cascading deletion

### LINKED ISSUE #15850: Django admin for version claims license is mandatory
### Describe the problem and steps to reproduce it:

I am trying to edit a version on the django admin page for versions, for exmaple setting or unsetting the `needs_human_review` flag.

### What happened?

I can't save the page because django complains that the _License_ field is required.

### What did you expect to happen?

Page should save even if the license is not set.

### Anything else we should know?
(Please include a link to the page, screenshots and any relevant files.)

### LINKED ISSUE #15876: Do not cascade deletion of a license to deletion of all versions using that license
### Describe the problem and steps to reproduce it:

STR:

1. Create a new license (via django admin)
2. Upload a version and select the license you just created.
3. Delete the license from the django admin

### What happened?

The uploaded version is also deleted.

### What did you expect to happen?

The uploaded version should be kept with the license attribute being set to null.

### Anything else we should know?
(Please include a link to the page, screenshots and any relevant files.)

### PR DESCRIPTION
**Pull request by wagnerand**
Fix #15850.
Fix #15876.

### CODE REVIEW

File: src/olympia/versions/models.py [hunk 1]
```diff
@@ -166,7 +166,7 @@ class Version(OnChangeMixin, ModelBase):
     addon = models.ForeignKey(
         'addons.Addon', related_name='versions', on_delete=models.CASCADE)
     license = models.ForeignKey(
-        'License', null=True, on_delete=models.CASCADE)
+        'License', null=True, blank=True, on_delete=models.CASCADE)
```
Comment (willdurand) [inline src/olympia/versions/models.py:169]: I am surprised that this change does not require a migration.

Comment (wagnerand) [reply src/olympia/versions/models.py:169]: Does https://docs.djangoproject.com/en/3.1/ref/models/fields/#django.db.models.Field.blank help?

Comment (willdurand) [reply src/olympia/versions/models.py:169]: No. Please run `./manage.py makemigrations versions` and edit the generated migration to only add the `AlterField` for the `license` field.

also cc @eviljeff, there are other operations (one of them being to use a `VersionStringField`

Comment (eviljeff) [reply src/olympia/versions/models.py:169]: 🤷‍♂️ the VersionStringField changes are no-ops at a db level

Comment (wagnerand) [reply src/olympia/versions/models.py:169]: @willdurand can you clarify what the expected migration should be? I ran `makemigrations` and the generated migrations file does not contain any changes related to patch:

```
# Generated by Django 2.2.16 on 2020-10-28 14:17

from django.db import migrations
import olympia.versions.fields


class Migration(migrations.Migration):

    dependencies = [
        ('versions', '0012_auto_20201020_1147'),
    ]

    operations = [
        migrations.RemoveField(
            model_name='version',
            name='recommendation_approved',
        ),
        migrations.AlterField(
            model_name='version',
            name='version',
            field=olympia.versions.fields.VersionStringField(default='0.1', max_length=255),
        ),
    ]
```

Comment (willdurand) [reply src/olympia/versions/models.py:169]: Interesting. Here is the one I got:

```python
  # Generated by Django 2.2.16 on 2020-10-28 13:55

  from django.db import migrations, models
  import django.db.models.deletion
  import olympia.versions.fields


  class Migration(migrations.Migration):

      dependencies = [
          ('versions', '0012_auto_20201020_1147'),
      ]

      operations = [
          migrations.RemoveField(
              model_name='version',
              name='recommendation_approved',
          ),
          migrations.AlterField(
              model_name='version',
              name='license',
              field=models.ForeignKey(blank=True, null=True, on_delete=django.db.models.deletion.CASCADE, to='versions.License'),
          ),
          migrations.AlterField(
              model_name='version',
              name='version',
              field=olympia.versions.fields.VersionStringField(default='0.1', max_length=255),
          ),
      ]
```

Comment (willdurand) [reply src/olympia/versions/models.py:169]: @eviljeff do you know when `recommendation_approved` was removed btw?

Comment (eviljeff) [reply src/olympia/versions/models.py:169]: https://github.com/mozilla/addons-server/commit/ca34f9c04e38dbf8f25e63e7ae074136bc4bf37e#diff-fdb00a545bc5f3b6d17bc710cde29232c9faee776dacf41c2448090cb6917b69

**Comment by willdurand** [comment]
If you rebase this patch and run `./manage.py makemigrations versions`, a migration should be generated.

**Comment by wagnerand** [comment]
@willdurand Rebased, thank you.

I also added a fix for https://github.com/mozilla/addons-server/issues/15876 since it's related.

**Review by willdurand — APPROVED** [review]

**Pull request merged**